tools/loadtesting/terraform/elasticsearch.tf

```diff
@@ -0,0 +1,178 @@
+data "aws_default_tags" "current" {}
+
+resource "aws_autoscaling_group" "elasticstack" {
+  name                      = "${local.prefix}-elasticstack"
+  max_size                  = 1
+  min_size                  = 1
+  health_check_grace_period = 3000
+  health_check_type         = "ELB"
+  desired_capacity          = 1
+  force_delete              = true
+  vpc_zone_identifier       = module.vpc.private_subnets
+  target_group_arns         = [aws_lb_target_group.elasticsearch.arn, aws_lb_target_group.elasticapm.arn, aws_lb_target_group.kibana.arn]
+
+  launch_template {
+    id      = aws_launch_template.elasticstack.id
+    version = "$Latest"
+  }
+
+  timeouts {
+    delete = "15m"
+  }
+
+  dynamic "tag" {
+    for_each = data.aws_default_tags.current.tags
+    content {
+      key                 = tag.key
+      value               = tag.value
+      propagate_at_launch = true
+    }
+  }
+
+  tag {
+    key                 = "ansible_repository"
+    value               = "https://github.com/fleetdm/fleet.git"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "ansible_playbook_path"
+    value               = "tools/loadtesting/terraform/elasticsearch_ansible"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "ansible_playbook_file"
+    value               = "elasticsearch.yml"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "ansible_branch"
+    value               = "zwinnerman-add-loadtest-infra-mine-fixup"
+    propagate_at_launch = true
+  }
```

Comment:
Yes, right now ES is in MVP mode basically. The end goal is to use ansible to maintain it since there are limits to user-data (it can only be so big) and any changes would require completely redeploying all instances. Ansible makes this much more maintainable. But since we don't have direct VPC access debugging this takes a long time, so in the interest of making APM available quickly I used a workable but less than ideal method of direct user-data stuff.

Reply:
The size limit is 16KB of base64 encoded data. Its fine for now, but once we start needing more advanced configuration with several services like SSM, managing individual users instead of just ec2-user, etc you quickly run out of bytes in user-data.

Reply:
It also means that we can standardize deployments for all EC2 stuff we might use in the future (for EKS if we want that, or ECS with spot instances, running mysql outside of RDS for loadtesting, etc).